src/app/sevices/task.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Task } from '../models/task';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService{
+  jsonSubscription: Subscription = Subscription.EMPTY;
+  tasksSubscription: Subscription = Subscription.EMPTY;
+  private tasksSubject: BehaviorSubject<Task[]> = new BehaviorSubject<Task[]>([]);
+  tasksObservable$: Observable<Task[]> = this.tasksSubject.asObservable();
+  tasksComplete$: BehaviorSubject<number> = new BehaviorSubject<number>(0);
```

Comment:
for behaviorsubjects you can just name it tasksCompleteSubj or tasksCompleteSubject